applications/pelion_client/configuration/nrf52840dk_nrf52840/app_ZRelease.conf

```diff
@@ -0,0 +1,121 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+################################################################################
+
+CONFIG_PELION_CLIENT_POWER_MANAGER_ENABLE=y
+CONFIG_PELION_CLIENT_POWER_MANAGER_STAY_ON=y
+
+CONFIG_PELION_CLIENT=y
+CONFIG_PELION_NRF_SECURITY=y
+CONFIG_PELION_UPDATE_DEVELOPER=y
+
+CONFIG_PELION_CLIENT_FACTORY_RESET_REQUEST_ENABLE=y
+
+################################################################################
+
+CONFIG_SIZE_OPTIMIZATIONS=y
+
+CONFIG_CAF=y
+CONFIG_CAF_BUTTONS=y
+CONFIG_CAF_BUTTONS_POLARITY_INVERSED=y
+
+CONFIG_CAF_LEDS=y
+CONFIG_PWM=y
+CONFIG_LED=y
+CONFIG_LED_PWM=y
+
+CONFIG_LINKER_ORPHAN_SECTION_PLACE=y
+
+CONFIG_NEWLIB_LIBC=y
+CONFIG_NEWLIB_LIBC_FLOAT_PRINTF=y
+CONFIG_CPLUSPLUS=y
+CONFIG_LIB_CPLUSPLUS=y
+CONFIG_POSIX_API=y
+
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=4096
+CONFIG_ISR_STACK_SIZE=2048
+CONFIG_MAIN_STACK_SIZE=4096
+
+CONFIG_HEAP_MEM_POOL_SIZE=16384
+
+CONFIG_SYS_CLOCK_TICKS_PER_SEC=32768
+
+CONFIG_FLASH=y
+CONFIG_FLASH_PAGE_LAYOUT=y
+CONFIG_FLASH_MAP=y
+
+CONFIG_PM_POLICY_APP=y
+
+# Network configuration
+CONFIG_NETWORKING=y
+CONFIG_NET_CONNECTION_MANAGER=y
+CONFIG_NET_SOCKETS=y
+CONFIG_NET_L2_OPENTHREAD=y
+
+# !!!
+# This has to be disabled for OpenThread!
+CONFIG_NET_IPV6_NBR_CACHE=n
+CONFIG_NET_IPV6_MLD=n
+# !!!
+
+CONFIG_DNS_RESOLVER=y
+CONFIG_DNS_SERVER_IP_ADDRESSES=y
+CONFIG_DNS_SERVER1="fdc5:f2ed:1010::ff:fe00:6400"
+CONFIG_DNS_SERVER2="fdde:dbb1:1de9::ff:fe00:c400"
+CONFIG_DNS_SERVER3="64:ff9b::4.4.4.4"
+
+# mbedTLS tuning
+CONFIG_MBEDTLS_HEAP_SIZE=32768
+CONFIG_MBEDTLS_ENTROPY_MAX_SOURCES=2
+
+# OpenThread related configuration
+CONFIG_OPENTHREAD_MBEDTLS=n
+CONFIG_OPENTHREAD_NRF_SECURITY=y
+CONFIG_OPENTHREAD_THREAD_STACK_SIZE=10240
+
+CONFIG_OPENTHREAD_JOINER=y
+CONFIG_OPENTHREAD_JOINER_AUTOSTART=y
+CONFIG_OPENTHREAD_JOINER_PSKD="THR3AD"
+
+CONFIG_OPENTHREAD_NORDIC_LIBRARY_MASTER=y
+
+CONFIG_OPENTHREAD_THREAD_VERSION_1_2=y
+CONFIG_OPENTHREAD_DUA=y
+CONFIG_OPENTHREAD_MLR=y
+CONFIG_OPENTHREAD_BACKBONE_ROUTER=y
+
+# Disable OpenThread autostart
+CONFIG_OPENTHREAD_MANUAL_START=y
+
+# The lines below are for easier debugging
+# (MPSL would make stepping through code hard as timing is very important there)
+CONFIG_NRF_802154_SL_OPENSOURCE=y
+CONFIG_NRF_802154_SL=n
+CONFIG_MPSL=n
```

Comment:
Do we need it if this is for easier debugging? From the top of my mind, I remember it was working withoud MPSL and without SL selected but then it hardfaults if breakpoint was hit.

Reply:
In release configurations there is no logging. The point is to create a configuration with minimum memory requirements to see on which devices the code has a chance of fitting.

Reply:
> Do we need it if this is for easier debugging?

I have not thought about this part but even without logs one could connect with gdb. It might help then.
I would also not enable it only for release configuration unless it is necessary (or helps to reduce memory footprint).